### PR TITLE
drivers: adc: support for Nuvoton numaker ADC

### PR DESCRIPTION
**Pull request by cyliangtw**
This PR is to add Nuvoton numaker series ADC controller.
It's verified on board: NuMaker-PFM-M467.

Depends on https://github.com/zephyrproject-rtos/hal_nuvoton/pull/8

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -188,7 +188,7 @@ manifest:
       groups:
         - hal
     - name: hal_nuvoton
-      revision: 584190e131655de1046088bd0d0735d83429ec7c
+      revision: pull/8/head
```
Comment (fabiobaltieri) [inline west.yml:191]: 68a91bb343ff47e40dbd9189a7d6e3ee801a7135

File: drivers/adc/adc_numaker.c [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+	}
+
+	if (sequence->buffer_size < needed_size) {
+		return -ENOBUFS;
+	}
+
+	return 0;
+}
+
+
```
Comment (fabiobaltieri) [inline drivers/adc/adc_numaker.c:107]: nit: just one blank line

File: tests/drivers/adc/adc_api/boards/numaker_pfm_m467.overlay [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+&pinctrl {
+/* EVB's UNO Pin A4 & D0 for channel 0 & 2 --> PB0, PB2 */
+	eadc0_default: eadc0_default {
+		group0 {
+			pinmux = <PB0MFP_EADC0_CH0>,
+				 <PB2MFP_EADC0_CH2>;
+		};
+	};
+};
+
```
Comment (fabiobaltieri) [inline tests/drivers/adc/adc_api/boards/numaker_pfm_m467.overlay:18]: Guess the comment want to be indented with the  node it refers to? plus it fits on 80 columns, no need to wrap it
```c
        /* EVB's UNO Pin A4 & D0 for channel 0 & 2 --> PB0, PB2 */              
        eadc0_default: eadc0_default {                                          
                group0 {                                                        
                        pinmux = <PB0MFP_EADC0_CH0>, <PB2MFP_EADC0_CH2>;        
                };                                                              
        };                                                                      
```

**Review by fabiobaltieri — COMMENTED** [review]
just nitpicks, good stuff

File: west.yml (hunk 1 above)
Comment (cyliangtw) [reply west.yml]: Done

File: tests/drivers/adc/adc_api/boards/numaker_pfm_m467.overlay (hunk 3 above)
Comment (cyliangtw) [reply tests/drivers/adc/adc_api/boards/numaker_pfm_m467.overlay:17]: Thanks and update it.

**Review by fabiobaltieri — DISMISSED** [review]

File: dts/bindings/adc/nuvoton,numaker-adc.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2023 Nuvoton Technology Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+description: Nuvoton, NuMaker PWM controller
```
Comment (ccli8) [inline dts/bindings/adc/nuvoton,numaker-adc.yaml:4]: typo

File: tests/drivers/adc/adc_api/boards/numaker_pfm_m467.overlay [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+&eadc0 {
+	status = "okay";
+	pinctrl-0 = <&eadc0_default>;
+	pinctrl-names = "default";
+	#address-cells = <1>;
+	#size-cells = <0>;
```
Comment (ccli8) [inline tests/drivers/adc/adc_api/boards/numaker_pfm_m467.overlay:23]: moved to m46x.dtsi?

File: dts/bindings/adc/nuvoton,numaker-adc.yaml (hunk 4 above)
Comment (cyliangtw) [reply dts/bindings/adc/nuvoton,numaker-adc.yaml]: Thanks and fixed the typo.

**Review by ccli8 — APPROVED** [review]
lgtm

**Review by fabiobaltieri — APPROVED** [review]

**Comment by cyliangtw** [comment]
Hi @ssekar15,
If you are available, please look into this PR.

**Review by ssekar15 — APPROVED** [review]
Look good for me, thanks.

**Pull request merged**